mesonbuild/compilers/detect.py

```diff
@@ -776,8 +785,7 @@ def detect_fortran_compiler(env: 'Environment', for_machine: MachineChoice) -> C
                     compiler, version, for_machine, is_cross, info,
                     full_version=full_version, linker=linker)
 
-            if 'flang' in out or 'clang' in out:
-                cls = fortran.FlangFortranCompiler
+            def _get_linker_try_windows(cls):
                 linker = None
                 if 'windows' in out or env.machines[for_machine].is_windows():
                     # If we're in a MINGW context this actually will use a gnu
```

Comment:
I broke this out into a separate function rather than duplicating it for both flang flavours. Not sure where this should ideally go. If it's not defined as a nested function, it would need a pretty extensive signature.